cmsRunCondor/cmsRunCondor.py

```diff
@@ -811,97 +840,121 @@ def cmsRunCondor(in_args=sys.argv[1:]):
 
-        status_filename = args.dag.replace(".dag", ".status")
-        write_dag_file(args.dag, status_filename, args.outputScript, total_num_jobs, job_name)
+        status_filename = args.dag.replace(".dag", "")  # TODO: handle if it doesn't end with .dag
+        status_filename += ".status"
+
+        cmsrun_dag = ht.DAGMan(filename=args.dag, status_file=status_filename)
 
     ###########################################################################
-    # submit to queue unless dry run
+    # Create Jobs
     ###########################################################################
-    if not args.dry:
-        if not args.dag:
-            subprocess.check_call(['condor_submit', args.outputScript])
+    script_dir = os.path.dirname(__file__)
 
+    cmsrun_jobs = ht.JobSet(
+        exe=os.path.join(script_dir, 'cmsRun_worker.sh'),
+        copy_exe=True,
+        filename=args.outputScript,
+        out_dir=args.log, out_file='cmsRun.$(cluster).$(process).out',
+        err_dir=args.log, err_file='cmsRun.$(cluster).$(process).err',
+        log_dir=args.log, log_file='cmsRun.$(cluster).$(process).log',
+        cpus=1, memory='2GB', disk='3GB',
+        # cpus=1, memory='1GB', disk='500MB',
+        certificate=True,
+        transfer_hdfs_input=True,
+        share_exe_setup=True,
+        common_input_files=[sandbox_local],  # EVERYTHING should be in the sandbox
+        hdfs_store=args.outputDir
+    )
+
+    output_files = get_output_files_from_config(args.config)
+
+    for job_ind in xrange(total_num_jobs):
+        # Construct args to pass to cmsRun_worker.sh on the worker node
+        args_dict = dict(output=args.outputDir, ind=job_ind)
+        report_filename = "report{ind}.xml".format(**args_dict)
+        args_dict['report'] = report_filename
+        args_str = "-o {output} -i {ind} -a $ENV(SCRAM_ARCH) " \
+                   "-c $ENV(CMSSW_VERSION) -r {report}".format(**args_dict)
+        if args.lumiMask or args.runRange:
+            if lumilist_filename:
+                args_str += ' -l ' + os.path.basename(lumilist_filename)
+            elif is_url(args.lumiMask):
+                args_str += ' -l ' + args.lumiMask
+        if args.useConfig:
+            args_str += ' -u'
+        if args.valgrind:
+            args_str += ' -m'
+        if args.callgrind:
+            args_str += ' -p'
+
+        # warning: this must be aligned with whatever cmsRun_worker.sh does...
+        job_output_files = [o.replace('.root', '_%d.root' % job_ind) for o in output_files]
+        job_output_files.append(report_filename)
+
+        if args.callgrind or args.valgrind:
+            job_output_files.append('callgrind.out.*')
+
+        job = ht.Job(
+            name='cmsRun_%d' % job_ind,
+            args=args_str,
+            input_files=None,
+            output_files=['CMSSW_*/src/'+ j for j in job_output_files],  # worker script operates in CMSSW/src
```

Comment:
In theory, yes. Although, since there should only be 1 CMSSW_* dir on the worker, it should be ok... And `job_output_files` takes the output file names from `process.TFileService` and `process.outputModules`, so is known at submission time.

What happens is `condor_worker.py` (running in `<worker>/scratch`) will finish running `cmsRun_worker.sh` (which has been running in `CMSSW_*/src`, and therefore ROOT files will be there). So then it looks for anything that satisfies `CMSSW_*/src/<filename>`, and if so, transfer it to the `outputDir`. I guess I should have used `os.environ['CMSSW_VERSION']` to actually specify the dir to look in. (but as a handy aside, it forced me to add in wildcard functionality to `htcondenser` 😄  )